### ISSUE TITLE
Use `Menu` instead of `Combobox` for search modal so we can navigate to results with `Enter`

### ISSUE THREAD

**Issue by jmduke** [buttondown/roadmap#2579]
I'm referring to the `headless-ui` components for both of these. No real UI changes needed, but someone should be able to navigate through results after typing.
